pkgs/test/lib/src/runner/wasm/platform.dart

```diff
@@ -0,0 +1,361 @@
+    }
+
+    cascade = cascade
+        .add(packagesDirHandler())
+        .add(_wasmHandler.handler)
+        .add(createStaticHandler(_root))
+        .add(_wrapperHandler);
+
+    var pipeline = shelf.Pipeline()
+        .addMiddleware(PathHandler.nestedIn(_secret))
+        .addHandler(cascade.handler);
+
+    _server.mount(shelf.Cascade()
+        .add(createFileHandler(faviconPath))
+        .add(pipeline)
+        .handler);
+  }
+
+  /// A handler that serves wrapper files used to bootstrap tests.
+  shelf.Response _wrapperHandler(shelf.Request request) {
+    var path = p.fromUri(request.url);
+
+    if (path.endsWith('.html')) {
+      var test = '${p.withoutExtension(path)}.dart';
+      var scriptBase = htmlEscape.convert(p.basename(test));
+      var link = '<link rel="x-dart-test" href="$scriptBase">';
+      var testName = htmlEscape.convert(test);
+      var template = _config.customHtmlTemplatePath ?? _defaultTemplatePath;
+      var contents = File(template).readAsStringSync();
+      var jsRuntime = 'dart2wasm_runtime.mjs';
+      var processedContents = contents
+          // Checked during loading phase that there is only one {{testScript}} placeholder.
+          .replaceFirst('{{testScript}}', link)
+          .replaceFirst('{{jsRuntimeUrl}}', jsRuntime)
+          .replaceFirst(
+              '{{wasmUrl}}', p.basename('$test.browser_test.dart.wasm'))
+          .replaceAll('{{testName}}', testName);
+      return shelf.Response.ok(processedContents,
+          headers: {'Content-Type': 'text/html'});
+    }
+
+    return shelf.Response.notFound('Not found.');
+  }
+
+  @override
+  ExecutableSettings parsePlatformSettings(YamlMap settings) =>
+      ExecutableSettings.parse(settings);
+
+  @override
+  ExecutableSettings mergePlatformSettings(
+          ExecutableSettings settings1, ExecutableSettings settings2) =>
+      settings1.merge(settings2);
+
+  @override
+  void customizePlatform(Runtime runtime, ExecutableSettings settings) {
+    var oldSettings =
+        _browserSettings[runtime] ?? _browserSettings[runtime.root];
+    if (oldSettings != null) settings = oldSettings.merge(settings);
+    _browserSettings[runtime] = settings;
+  }
+
+  /// Loads the test suite at [path] on the platform [platform].
+  ///
+  /// This will start a browser to load the suite if one isn't already running.
+  /// Throws an [ArgumentError] if `platform.platform` isn't a browser.
+  @override
+  Future<RunnerSuite?> load(String path, SuitePlatform platform,
+      SuiteConfiguration suiteConfig, Map<String, Object?> message) async {
+    if (suiteConfig.precompiledPath != null) {
+      throw UnsupportedError(
+          'The wasm platform doesn\'t support precompiled suites');
+    }
+
+    var browser = platform.runtime;
+    assert(suiteConfig.runtimes.contains(browser.identifier));
+
+    if (!browser.isBrowser) {
+      throw ArgumentError('$browser is not a browser.');
+    }
+
+    // TODO: Support custom html?
```

Comment:
Remove?

Reply:
I think this one is valid - we likely will want to support custom  html at some point